### PR TITLE
feat(testing): adding e2e cypress tests for Dropdown

### PR DESCRIPTION
**Pull request by silviuaavram**
Added end to end tests using Cypress. Also added them to CircleCI as a separate job. At the moment there are just a few `Dropdown` tests.

Added cypress as a gulp task, and used that in package.json as `yarn test:2e2`.

Feedback on every part of this PR is welcomed.

### CODE REVIEW

**Review by kolaps33 — APPROVED** [review]

File: .circleci/config.yml [hunk 1]
```diff
@@ -140,3 +148,41 @@ jobs:
...
+            curl -o- -L https://yarnpkg.com/install.sh | bash
+            echo 'export PATH="${PATH}:${HOME}/.yarn/bin"' >> $BASH_ENV
+      - checkout
+      # because we don't invoke npm (we use yarn) we need to add npm bin to PATH manually
+      - run:
+          name: Add npm bin to PATH
+          command: echo 'export PATH="${PATH}:$(npm bin)"' >> $BASH_ENV
+      - restore_cache:
+          keys:
+            - v1-dependencies-{{ checksum "yarn.lock" }}
```
Comment (miroslavstastny) [inline .circleci/config.yml]: This changed to `v1.1` in master, please update it here (and in `save_cache`) as well

**Review by miroslavstastny — COMMENTED** [review]
The code looks good 👍 

The problem I have is that it adds the third parallel process to be run in CI. Considering we have 4 processes limit in our plan we can no longer run two builds in parallel. Does it make sense to have this as a separate process? Or as `t(screener)+t(cypress) < t(build)` can we make this a part of visual tests process? @kuzhelov what's your opinion on this?

**Comment by miroslavstastny** [comment]
If we decide to merge the current approach with a separate process, we should also mark `ci/circleci: e2e` as required in branch protection rules for `master` in github config.

File: .circleci/config.yml (hunk 1 above)
Comment (silviuaavram) [reply .circleci/config.yml]: thanks for pointing out, changed

**Pull request closed without merging**